### PR TITLE
Add (vector)postprocessors and snip markers to dump problem

### PR DESCRIPTION
**Pull request by dschwen**
I have an Atom plugin ready waiting for these changes:
https://github.com/dschwen/action-explode-moose

Refs #8875

### CODE REVIEW

File: framework/src/problems/DumpObjectsProblem.C [hunk 1]
```diff
@@ -276,8 +294,10 @@ DumpObjectsProblem::dumpGeneratedSyntax(const std::string path)
   if (pathit == _generated_syntax.end())
     return;
 
+  Moose::out << "**START DUMP DATA**\n";
```
Comment (loganharbour) [inline framework/src/problems/DumpObjectsProblem.C:297]: Is this here because you're parsing the console output in the plugin?

**Review by loganharbour — DISMISSED** [review]

File: framework/include/problems/DumpObjectsProblem.h [hunk 2]
```diff
@@ -76,6 +76,12 @@ class DumpObjectsProblem : public FEProblemBase
   void addMaterial(const std::string & type,
                    const std::string & name,
                    InputParameters & parameters) override;
+  void addPostprocessor(std::string type,
```
Comment (permcody) [inline framework/include/problems/DumpObjectsProblem.h:79]: minor nitpick: You might make these const references too like the other declarations.

File: framework/src/problems/DumpObjectsProblem.C (hunk 1 above)
Comment (dschwen) [reply framework/src/problems/DumpObjectsProblem.C:297]: Yep

File: framework/include/problems/DumpObjectsProblem.h (hunk 2 above)
Comment (dschwen) [reply framework/include/problems/DumpObjectsProblem.h:79]: Yeah, I agree, this was an oversight.

**Review by loganharbour — APPROVED** [review]

**Pull request merged**